### ISSUE TITLE
App crashes after connected on iOS with Flutter 3.0 stable

### ISSUE THREAD

**Issue by acgacgacgacg** [livekit/client-sdk-flutter#118]
## Problem

After upgraded Flutter to 3.0 stable, the app crashes each time after connected to the server.
This occurred on iphoneX device with iOS 15.4.

## Log

```bash
flutter: 09:22:15: [EngineEvent] Engine#1052307608 EngineSubscriberPeerStateUpdatedEvent(state: RTCPeerConnectionState.RTCPeerConnectionStateConnecting, isPrimary: true)
flutter: 09:22:15: pc.getRemoteDescription Instance of 'RTCSessionDescription'
flutter: 09:22:16: [EngineEvent] Engine#1052307608 EngineSubscriberPeerStateUpdatedEvent(state: RTCPeerConnectionState.RTCPeerConnectionStateConnected, isPrimary: true)
flutter: 09:22:16: EventsEmitter<EngineEvent>#211189967 event was cancelled by func
flutter: 09:22:16: Engine ConnectionState connecting -> connected
flutter: 09:22:16: [EngineEvent] Engine#1052307608 EngineConnectionStateUpdatedEvent(newState: connected, didReconnect: false, disconnectReason: null)
[VERBOSE-3:platform_message_handler_ios.mm(106)] Check failed: task_runners_.GetPlatformTaskRunner()->RunsTasksOnCurrentThread().
* thread #24, name = 'signaling_thread 0x0x155f28300', stop reason = signal SIGABRT
    frame #0: 0x00000001bbf79bbc libsystem_kernel.dylib`__pthread_kill + 8
libsystem_kernel.dylib`__pthread_kill:
->  0x1bbf79bbc <+8>:  b.lo   0x1bbf79bd8               ; <+36>
    0x1bbf79bc0 <+12>: stp    x29, x30, [sp, #-0x10]!
    0x1bbf79bc4 <+16>: mov    x29, sp
    0x1bbf79bc8 <+20>: bl     0x1bbf7560c               ; cerror_nocancel
Target 0: (Runner) stopped.
Lost connection to device.
```

## How to reproduce

Use the example app and build with Flutter 3.0 stable.

## flutter doctor

```bash
flutter doctor -v
[✓] Flutter (Channel stable, 3.0.0, on macOS 12.3.1 21E258 darwin-x64, locale zh-Hans-JP)
    • Flutter version 3.0.0 at /Users/aldin/development/flutter
    • Upstream repository git@github.com:flutter/flutter.git
    • Framework revision ee4e09cce0 (6 天前), 2022-05-09 16:45:18 -0700
    • Engine revision d1b9a6938a
    • Dart version 2.17.0
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/aldin/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] VS Code (version 1.67.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.40.0
```

**Comment by rokk4** [comment]
Same here https://github.com/livekit/client-sdk-flutter/issues/116#issuecomment-1127045489

**Comment by hiroshihorie** [comment]
Looking into this one

**Comment by hiroshihorie** [comment]
This PR makes it not crash anymore 👉 https://github.com/flutter-webrtc/flutter-webrtc/pull/953
Please wait until it gets merged, then we can update our package.

**Comment by hiroshihorie** [comment]
This should be fixed in v1.0
